control/control_plane.go

```diff
@@ -746,38 +747,10 @@ func (c *ControlPlane) Serve(readyChan chan<- bool, listener *Listener) (err err
 				pktDst := RetrieveOriginalDest(oob)
 				routingResult, err := c.core.RetrieveRoutingResult(src, pktDst, unix.IPPROTO_UDP)
 				if err != nil {
-					// WAN. Old method.
-					lastErr := err
-					addrHdr, dataOffset, err := ParseAddrHdr(data)
-					if err != nil {
-						if c.tproxyPortProtect {
-							c.log.Warnf("No AddrPort presented: %v, %v", lastErr, err)
-							return
-						} else {
-							routingResult = &bpfRoutingResult{
-								Mark:     0,
-								Must:     0,
-								Mac:      [6]uint8{},
-								Outbound: uint8(consts.OutboundControlPlaneRouting),
-								Pname:    [16]uint8{},
-								Pid:      0,
-								Dscp:     0,
-							}
-							realDst = pktDst
-							goto destRetrieved
-						}
-					}
-					data = data[dataOffset:]
-					routingResult = &addrHdr.RoutingResult
-					__ip := common.Ipv6Uint32ArrayToByteSlice(addrHdr.Ip)
-					_ip, _ := netip.AddrFromSlice(__ip)
-					// Comment it because them SHOULD equal.
-					//src = netip.AddrPortFrom(_ip, src.Port())
-					realDst = netip.AddrPortFrom(_ip, addrHdr.Port)
+					c.log.Warnf("No AddrPort presented: %v", err)
```

Comment:
return from here because we cannot handle this situation.
```
return fmt.Errorf("failed to retrieve target info %v: %v", pktDst.String(), err)
```

Reply:
This is inside a goroutine without return values, can't return an error LOL

Reply:
> This is inside a goroutine without return values, can't return an error LOL

Got it. However, we should return after log and stop to handle this packet.